db/migrate/20220425082348_add_cohorts.rb

```diff
@@ -0,0 +1,142 @@
+class AddCohorts < ActiveRecord::Migration[6.1]
+  class Startup < ApplicationRecord
+    belongs_to :level
+    has_many :founders, dependent: :restrict_with_error
+    has_many :faculty_startup_enrollments, dependent: :destroy
+    has_many :faculty, through: :faculty_startup_enrollments
+    has_one :course, through: :level
+  end
+
+  class Founder < ApplicationRecord
+    belongs_to :startup
+    has_one :level, through: :startup
+    has_one :course, through: :level
+  end
+
+  class Level < ApplicationRecord
+    has_many :startups, dependent: :restrict_with_error
+    belongs_to :course
+  end
+
+  class Course < ApplicationRecord
+    has_many :levels, dependent: :restrict_with_error
+    has_many :startups, through: :levels
+    has_many :founders, through: :startups
+    has_many :faculty_course_enrollments, dependent: :destroy
+    has_many :faculty, through: :faculty_course_enrollments
+  end
+
+  class Cohort < ApplicationRecord
+  end
+
+  class FacultyCourseEnrollment < ApplicationRecord
+    belongs_to :faculty
+    belongs_to :course
+  end
+
+  class FacultyStartupEnrollment < ApplicationRecord
+    belongs_to :faculty
+    belongs_to :founder
+  end
+
+  class FacultyCohortEnrollment < ApplicationRecord
+    belongs_to :faculty
+    belongs_to :cohort
+  end
+
+  class FacultyFounderEnrollment < ApplicationRecord
+    belongs_to :faculty
+    belongs_to :founder
+  end
+
+  class Faculty < ApplicationRecord
+    has_many :faculty_course_enrollments, dependent: :destroy
+    has_many :courses, through: :faculty_course_enrollments
+    has_many :faculty_startup_enrollments, dependent: :destroy
+    has_many :founders, through: :faculty_startup_enrollments
+  end
+
+  def up
+    create_table :cohorts do |t|
+      t.string :name
+      t.string :description
+      t.datetime :ends_at
+
+      t.timestamps
+    end
+
+    add_reference :courses, :default_cohort, foreign_key: { to_table: :cohorts }
```

Comment:
I had a discussion with @vinutv about this, and we think we might be able to avoid this _default_cohort_ selection.

Instead of assigning a default cohort, it should be sufficient to add a cohort selection on the form which converts an applicant into a student. There's a drawer right now that pops up asking for additional info right before an applicant is onboarded as a student. This form could be updated to include a cohort selection.

The other use of the default cohort that we discussed was a default selection in the _Add Students_ form - this can be safely ignored, I think.

Reply:
This was supposed to be used for public signup. If you don't have a processing url, students would get onboarded automatically upon email verification which would need a default cohort.

Reply:
Understood!